test/publish.test.js

```diff
@@ -101,6 +101,7 @@ test.serial('Publish a release with one asset', async t => {
   };
   const nextRelease = {version: '1.0.0', gitHead: '123', gitTag: 'v1.0.0', notes: 'Test release note body'};
   const options = {branch: 'master', repositoryUrl: `https://github.com/${owner}/${repo}.git`};
+  const untaggedReleaseUrl = `https://github.com/${owner}/${repo}/releases/untagged`;
```

Comment:
just a minor, but the URL returned by github ends with `-[commit hash]`, so I would suggest we do the same here

```suggestion
  const untaggedReleaseUrl = `https://github.com/${owner}/${repo}/releases/untagged-123`;
```

And adopt the tests accordingly. Other than that this is good to go :)